athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/AwsRestHighLevelClient.java

```diff
@@ -150,17 +153,17 @@ public Set<Integer> getShardIds(String index, long timeout)
         ClusterHealthResponse response = cluster().health(request, RequestOptions.DEFAULT);
 
         if (response.isTimedOut()) {
-            throw new RuntimeException("Request timed out for index (" + index + ").");
+            throw new AthenaConnectorException("Request timed out for index (" + index + ").", new ErrorDetails().withErrorCode(FederationSourceErrorCode.OperationTimeoutException.toString()));
         }
         else if (response.getActiveShards() == 0) {
-            throw new RuntimeException("There are no active shards for index (" + index + ").");
+            throw new AthenaConnectorException("There are no active shards for index (" + index + ").", new ErrorDetails().withErrorCode(FederationSourceErrorCode.InternalServiceException.toString()));
         }
         else if (response.getStatus() == ClusterHealthStatus.RED) {
```

Comment:
How are we handling access denied?

Reply:
AwsRestHighLevelClient only returns the client. the client will be returning a response in ElasticsearchMetadataHandler. so I think that will throw access denied on the metadata handler class as there is not any exclusive exception thrown.